src/metabase/api/dashboard.clj

```diff
@@ -784,38 +786,63 @@
   "How many results to return when chain filtering"
   1000)
 
-(mu/defn ^:private mappings->field-ids :- [:maybe [:set ms/PositiveInt]]
+(defn- find-template-tag
+  "Fetch the `:field` clause from `dashcard` referenced by `template-tag`.
+
+    (find-template-tag [:template-tag :company] some-dashcard) ; -> [:field 100 nil]"
+  [dimension card]
+  (when-let [tag (first (mbql.u/get-clause :template-tag dimension))]
+    (get-in card [:dataset_query :native :template-tags (u/qualified-name tag)])))
+
+(mu/defn ^:private mappings->fields :- [:maybe [:set [:map
+                                                      [:field-id ms/PositiveInt]
+                                                      [:options [:maybe map?]]]]]
   [parameter-mappings :- [:maybe [:or [:set dashboard-card/ParamMapping] [:sequential dashboard-card/ParamMapping]]]]
-  (set (for [{{:keys [card]} :dashcard :keys [target]} parameter-mappings
-             :let  [field-clause (params/param-target->field-clause target card)]
-             :when field-clause
-             :let  [{:keys [result_metadata]} card
-                    field-id (or
-                              ;; Get the field id from the field-clause if it contains it. This is the common case for
-                              ;; mbql queries.
-                              (mbql.u/match-one field-clause [:field (id :guard integer?) _] id)
-                              ;; Attempt to get the field clause from the model metadata corresponding to the field.
-                              ;; This is the common case for native queries in which mappings from original columns
-                              ;; have been performed using model metadata.
-                              (:id (qp.util/field->field-info field-clause result_metadata)))]
+  (set (for [{:keys [target] {:keys [card]} :dashcard} parameter-mappings
+             :let  [dimension (->> (mbql.normalize/normalize-tokens target :ignore-path)
+                                   (mbql.u/get-clause :dimension)
+                                   first)]
+             :when dimension
+             :let  [ttag      (find-template-tag dimension card)
```

Comment:
can you show me the e2e test that makes template tag has options? 
if it's not possible on the UI then let's just get rid of this so this code is less confusing. that's way we can make this to function to return a set of ids